### PR TITLE
Add support for syscall_unlinkat

### PR DESCRIPTION
**Pull request by Sahakait**

### CODE REVIEW

File: kernel/syscall.c
Comment (Sahakait) [inline kernel/syscall.c:1559]: Required to pass ltp

File: kernel/libc.c
Comment (Francis-Liu) [inline kernel/libc.c:534]: No need to change libc.c

File: tests/ltp/hostfs_tests_passed.txt
Comment (Francis-Liu) [inline tests/ltp/hostfs_tests_passed.txt:109]: Please try to enable the same test case in all the following files:

ext2fs_fs_tests_passed.fork.txt  ext2fs_tests_passed.fork.txt  hostfs_fs_tests_passed.fork.txt  hostfs_tests_passed.fork.txt  ramfs_fs_tests_passed.fork.txt  ramfs_tests_passed.fork.txt
ext2fs_fs_tests_passed.txt       ext2fs_tests_passed.txt       hostfs_fs_tests_passed.txt       hostfs_tests_passed.txt       ramfs_fs_tests_passed.txt       ramfs_tests_passed.txt

**Comment by Francis-Liu** [comment]
I have a pending PR #510 which rewrites `myst_get_absolute_path_from_dirfd`. Please hold this PR until #510 is merged.

**Comment by Francis-Liu** [comment]
Please update `syscall-limitations.md` accordingly.

**Comment by Francis-Liu** [comment]
Please update `https://github.com/deislabs/mystikos/tree/main/tests/ltp#test-matrix` accordingly.

File: kernel/libc.c
Comment (Sahakait) [reply kernel/libc.c]: Change has been reverted. 

Context: `libc.c` is used for gcov, no additional syscall support needed.

**Comment by Sahakait** [comment]
Test matrix and list of passing test cases have been updated.

File: kernel/syscall.c
Comment (jxyang) [inline kernel/syscall.c:1648]: I am skeptical that we can't pass in empty or null pathname to unlinkat. Could you find out by writing a test program and run with Linux?

Comment (jxyang) [reply kernel/syscall.c:1648]: I am guess `unlinkat` would remove the folder `dirfd` if `AT_REMOVEDIR` is specified.

Comment (Francis-Liu) [reply kernel/syscall.c:1648]: @Sahakait I found a post regarding empty pathname: https://bugzilla.kernel.org/show_bug.cgi?id=93441. It seems if pathname is an empty string, then flags must add `AT_EMPTY_PATH`. If that's the case, then the following checks are covered by `myst_get_absolute_path_from_dirfd`. You could remove the check here.

https://github.com/deislabs/mystikos/blob/e33dcd868fb0c06c8534f06f89cdc599158c97f5/kernel/syscall.c#L873-L874

https://github.com/deislabs/mystikos/blob/e33dcd868fb0c06c8534f06f89cdc599158c97f5/kernel/syscall.c#L881-L884

Comment (jxyang) [reply kernel/syscall.c:1648]: Thanks Francis. The issue you referred to is still open, meaning the Linux kernel hasn't implemented `AT_EMPTY_PATH` for `unlinkat` yet. I will resolve this comment.

**Review by jxyang — APPROVED** [review]

**Comment by jxyang** [comment]
Please rebase to the main branch and run pipeline again before merging.

File: kernel/syscall.c
Comment (Francis-Liu) [inline kernel/syscall.c:1649]: Please remove these 2 check as they are covered by `myst_get_absolute_path_from_dirfd`.

**Review by Francis-Liu — APPROVED** [review]
LGTM. I saw 3 merge commits. Please consider use rebase to cleanup commit history next time.

**Comment by jxyang** [comment]
Still showing merge conflicts. Try rebasing with the main branch of the original repo, not your forked repo.

**Review by Francis-Liu — APPROVED** [review]

**Pull request merged**
